apps/advisor/pkg/app/checks/plugincheck/check.go

```diff
@@ -0,0 +1,83 @@
+package plugincheck
+
+import (
+	"context"
+	"fmt"
+	sysruntime "runtime"
+
+	"github.com/Masterminds/semver/v3"
+	advisor "github.com/grafana/grafana/apps/advisor/pkg/apis/advisor/v0alpha1"
+	"github.com/grafana/grafana/apps/advisor/pkg/app/checks"
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/services"
+	"github.com/grafana/grafana/pkg/plugins/repo"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginstore"
+)
+
+func New(
+	pluginStore pluginstore.Store,
+	pluginRepo repo.Service,
+) checks.Check {
+	return &check{
+		PluginStore: pluginStore,
+		PluginRepo:  pluginRepo,
+	}
+}
+
+type check struct {
+	PluginStore pluginstore.Store
+	PluginRepo  repo.Service
+}
+
+func (c *check) Type() string {
+	return "plugin"
+}
+
+func (c *check) Run(ctx context.Context, _ *advisor.CheckSpec) (*advisor.CheckV0alpha1StatusReport, error) {
+	ps := c.PluginStore.Plugins(ctx)
+
+	errs := []advisor.CheckV0alpha1StatusReportErrors{}
+	for _, p := range ps {
+		// Check if plugin is deprecated
+		i, err := c.PluginRepo.PluginInfo(ctx, p.ID)
+		if err != nil {
+			continue
+		}
+		if i.Status == "deprecated" {
+			errs = append(errs, advisor.CheckV0alpha1StatusReportErrors{
+				Severity: advisor.CheckStatusSeverityHigh,
+				Reason:   fmt.Sprintf("Plugin deprecated: %s", p.ID),
+				Action:   "Look for alternatives",
+			})
+		}
+
+		// Check if plugin has a newer version
+		compatOpts := repo.NewCompatOpts(services.GrafanaVersion, sysruntime.GOOS, sysruntime.GOARCH)
```

Comment:
I guess we only want to perform this check when the plugin is:
- not managed
- not preinstalled with a specified version
- not a core plugin?
- {anything else I missed}

Reply:
true, I'll add a check for that